webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js

```diff
@@ -197,44 +197,59 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
     return this._i18n;
   }
 
-  _setup() {
+  _closeGrader(e) {
 
-    //Disable Offcanvas FocusTrap
-    bootstrap.Offcanvas.prototype._initializeFocusTrap = function () { return { activate() {}, deactivate() {} }; };
+    this.querySelector("sakai-rubric-grading")?.closeCommentEditors();
 
-    this.feedbackCommentEditor = this._replaceWithEditor("grader-feedback-comment", data => {
-      this._submission.feedbackComment = data;
-      this._gradeOrCommentsModified = true;
-    });
+    if (this.modified || this.querySelector("sakai-grader-file-picker")?.hasFiles()) {
+      e.preventDefault();
+      this._save({ bannerTimout: 2000 });
+    }
 
-    this.privateNotesEditor = this._replaceWithEditor("grader-private-notes", data => {
-      this._submission.privateNotes = data;
-      this._gradeOrCommentsModified = true;
-    });
+    // Close all the collapses on the hidden event, so we don't have loads of sliding
+    // about going on at once.
+    bootstrap.Collapse.getInstance(this.querySelector("#feedback-block"))?.hide();
+    bootstrap.Collapse.getInstance(this.querySelector("#private-notes-block"))?.hide();
+    this._feedbackCommentRemoved = false;
+    this._privateNotesRemoved = false;
+    this._closeRubric();
+    this._closeStudentRubric();
 
-    document.getElementById("grader").addEventListener("hide.bs.offcanvas", e => {
+    bootstrap.Collapse.getInstance(this.querySelector("#feedback-block"))?.hide();
 
-      this.querySelector("sakai-rubric-grading")?.closeCommentEditors();
+    this._toggleGrader();
+  }
 
-      if (this.modified || this.querySelector("sakai-grader-file-picker")?.hasFiles()) {
-        e.preventDefault();
-        this._save({ closeSidebarTimeout: 2000 });
-      }
+  _toggleGrader() {
 
-    });
+    const graderEl = this.querySelector("#grader");
+    const gradableEl = this.querySelector("#grader-gradable-container");
 
-    document.getElementById("grader").addEventListener("hidden.bs.offcanvas", () => {
+    if (graderEl.classList.contains("d-none")) {
+      graderEl.classList.remove("d-none");
+      graderEl.classList.add("d-block");
+      gradableEl.classList.add("d-none");
+      graderEl.classList.add("rounded-3");
+      graderEl.classList.add("mt-2");
+    } else {
+      graderEl.classList.remove("d-block");
+      graderEl.classList.remove("rounded-3");
+      graderEl.classList.add("d-none");
+      gradableEl.classList.remove("d-none");
+      graderEl.classList.remove("mt-2");
+    }
+  }
```

Comment:
maybe we can make it simple

Reply:
We could probably. Make it a single class in library maybe.